contracts/v2/marketplace/TokenMarketplaceV2.sol

```diff
@@ -0,0 +1,496 @@
+    address _artistAccount,
+    uint256 _artistCommissionRate,
+    address _optionalCommissionRecipient,
+    uint256 _optionalCommissionRate,
+    uint256 _remainingRoyalties
+  ) internal {
+    uint256 _totalCollaboratorsRate = _artistCommissionRate.add(_optionalCommissionRate);
+    uint256 _scaledUpCommission = _artistCommissionRate.mul(10 ** 18);
+
+    // work out % of royalties total to split e.g. 43 / 85 = 50.5882353%
+    uint256 primaryArtistPercentage = _scaledUpCommission.div(_totalCollaboratorsRate);
+
+    uint256 totalPrimaryRoyaltiesToArtist = _remainingRoyalties.mul(primaryArtistPercentage).div(10 ** 18);
+    _artistAccount.transfer(totalPrimaryRoyaltiesToArtist);
+
+    uint256 remainingRoyaltiesToCollaborator = _remainingRoyalties.sub(totalPrimaryRoyaltiesToArtist);
+    _optionalCommissionRecipient.transfer(remainingRoyaltiesToCollaborator);
+  }
+
+  ///////////////////
+  // Query Methods //
+  ///////////////////
+
+  function tokenOffer(uint256 _tokenId) external view returns (address _bidder, uint256 _offer, address _owner, bool _enabled, bool _paused) {
+    Offer memory offer = offers[_tokenId];
+    return (
+    offer.bidder,
+    offer.offer,
+    kodaAddress.ownerOf(_tokenId),
+    !disabledTokens[_tokenId],
+    paused
+    );
+  }
+
+  function determineSaleValues(uint256 _tokenId) external view returns (uint256 _sellerTotal, uint256 _platformFee, uint256 _royaltyFee) {
+    Offer memory offer = offers[_tokenId];
+    uint256 offerValue = offer.offer;
+    uint256 fee = offerValue.div(1000).mul(platformFeePercentage);
+    uint256 royalties = offerValue.div(1000).mul(artistRoyaltyPercentage);
+
+    return (
+    offer.offer.sub(fee).sub(royalties),
+    fee,
+    royalties
+    );
+  }
+
+  function tokenListingDetails(uint256 _tokenId) external view returns (uint256 _price, address _lister, address _currentOwner) {
+    Listing memory listing = listings[_tokenId];
+    return (
+    listing.price,
+    listing.seller,
+    kodaAddress.ownerOf(_tokenId)
+    );
+  }
+
+  function isContract(address account) internal view returns (bool) {
+    // This method relies in extcodesize, which returns 0 for contracts in
+    // construction, since the code is only stored at the end of the
+    // constructor execution.
+    uint256 size;
+    // solhint-disable-next-line no-inline-assembly
+    assembly {size := extcodesize(account)}
+    return size > 0;
+  }
+
+  ///////////////////
+  // Admin Actions //
+  ///////////////////
+
+  function disableAuction(uint256 _tokenId)
+  public
+  onlyIfWhitelisted(msg.sender)
+  {
+    _refundHighestBidder(_tokenId);
+    _delistToken(_tokenId);
+
+    disabledTokens[_tokenId] = true;
+
+    emit AuctionDisabled(_tokenId, msg.sender);
+  }
+
+  function enableAuction(uint256 _tokenId)
+  public
+  onlyIfWhitelisted(msg.sender)
+  {
+    _refundHighestBidder(_tokenId);
+
+    disabledTokens[_tokenId] = false;
+
+    emit AuctionEnabled(_tokenId, msg.sender);
+  }
```

Comment:
Do we need a reason/function to delist / disabled a listing? Probably not...

Reply:
Added